src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -628,6 +629,82 @@ static term nif_crypto_hash(Context *ctx, int argc, term argv[])
     return term_from_literal_binary(digest, digest_len, &ctx->heap, ctx->global);
 }
 
+static const AtomStringIntPair cipher_table[] = {
+    { ATOM_STR("\xB", "aes_128_cbc"), MBEDTLS_CIPHER_AES_128_CBC },
+    { ATOM_STR("\xB", "aes_128_ctr"), MBEDTLS_CIPHER_AES_128_CTR },
+    SELECT_INT_DEFAULT(MBEDTLS_CIPHER_NONE)
+};
+
+static term nif_crypto_crypto_one_time(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    term cipher_term = argv[0];
+    mbedtls_cipher_type_t cipher
+        = interop_atom_term_select_int(cipher_table, cipher_term, ctx->global);
+    if (UNLIKELY(cipher == MBEDTLS_CIPHER_NONE)) {
+        RAISE_ERROR(BADARG_ATOM);
```

Comment:
crypto module now uses 3-tuple error handling. Tests would allow us to ensure we are compatible.

Reply:
ok, going to fix this.